source/extensions/filters/common/local_ratelimit/local_ratelimit_impl.cc

```diff
@@ -145,6 +152,39 @@ OptRef<const LocalRateLimiterImpl::LocalDescriptorImpl> LocalRateLimiterImpl::de
 
 bool LocalRateLimiterImpl::requestAllowed(
     absl::Span<const RateLimit::LocalDescriptor> request_descriptors) const {
+  if (Runtime::runtimeFeatureEnabled(
+          "envoy.reloadable_features.http_local_ratelimit_match_all_descriptors")) {
+    // If a request is limited by a descriptor, it should not consume token from the remaining
+    // matched descriptors, so we first check the remaining tokens of matched descriptors instead of
+    // trying to consuming their tokens. If one of these tokens is 0, it just return false and will
+    // not consume any tokens of matched descriptors. Otherwise, it will try to consume 1 token from
+    // each matched descriptor.
+    bool allow = hasEnoughTokens(tokens_);
+    // Global token is not enough.
+    if (!allow) {
+      return allow;
+    }
+    std::vector<TokenState*> token_states;
+    if (!descriptors_.empty() && !request_descriptors.empty()) {
+      for (const auto& request_descriptor : request_descriptors) {
+        if (auto it = descriptors_.find(request_descriptor); it != descriptors_.end()) {
+          allow &= hasEnoughTokens(*it->token_state_);
+          // Descriptor token is not enough.
+          if (!allow) {
+            return allow;
+          }
+          token_states.push_back(it->token_state_.get());
+        }
+      }
+    }
+
+    // All tokens are enough.
```

Comment:
Per above this could change between then and now, so I'm not sure the point of doing the then check. :)